unix_integration/src/selinux_util.rs

```diff
@@ -1,9 +1,12 @@
 use std::ffi::CString;
 
-use selinux::{kernel_support, label::back_end::File, label::Labeler, KernelSupport};
+use selinux::{
+    current_mode, kernel_support, label::back_end::File, label::Labeler, KernelSupport, SELinuxMode,
+};
 
 pub fn supported() -> bool {
-    return !matches!(kernel_support(), KernelSupport::Unsupported);
+    return !matches!(kernel_support(), KernelSupport::Unsupported)
```

Comment:
As pedantic as it seems, it may be better to list the conditions we *do* need for support instead. Perhaps this should be `Supported && ( Permissive || Enforcing )` or similar so that the logic reads as "this is supported if kernel support is on and the mode is permissive or enforcing".

Reply:
Sure, though the `KernelSupport` enum can consist of `SELinux` and `SELinuxMLS` which may be confusing by itself I think. Maybe we should just expand the `SELinuxMode` check? If anything we can also add a comment to clarify what exactly we are checking here.

Reply:
Yeah, the comments would help. Alternately we could do a match or something?

Reply:
> Alternately we could do a match or something?

That's another option. I will experiment and see what I can come up with. :)

Reply:
Thanks, I think it would be really good to be clear and explicit here is all :) just in case we expand the modes we support later or have to change it.

Reply:
I think I'm happy with this. I would have liked to just use match arms to check `NotRunning`, `Permissive`, and `Enforcing`, but the `SELinuxMode` enum is `non_exhaustive` for some reason which means we must account for `_`.